src/ln/peer_handler.rs

```diff
@@ -312,7 +316,7 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 	/// course of this function!
 	///
 	/// Panics if the descriptor was not previously registered in a new_*_connection event.
-	pub fn read_event(&self, peer_descriptor: &mut Descriptor, data: Vec<u8>) -> Result<bool, PeerHandleError> {
+	pub fn read_event(&mut self, peer_descriptor: &mut Descriptor, data: Vec<u8>) -> Result<bool, PeerHandleError> {
```

Comment:
Hmm, can we do without the mut? mut means we cant call something from multiple threads simultaneously from clients, which really makes this not-so-useful. I don't actually see what needs mut here anyway?

Reply:
I believe this might be a left over from a previous commit